component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPAuthenticatorCredentials.java

```diff
@@ -398,7 +398,8 @@ private void storeSecretKey(String secretKey, String username) {
 			UserRealm userRealm = TOTPUtil.getUserRealm(username);
 			if (userRealm != null) {
 				userClaims.put(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL, TOTPUtil.encrypt(secretKey));
-				userRealm.getUserStoreManager().setUserClaimValues(username, userClaims, null);
+				userRealm.getUserStoreManager().setUserClaimValues(MultitenantUtils.getTenantAwareUsername(username),
```

Comment:
Will this cause an issue in a scenario like email as username?

Reply:
Also can we look at refactoring this method a bit. 
(Take an overall look at authorizeAndStoreSecret() and storeSecretKey() methods)

Reply:
`Will this cause an issue in a scenario like email as username?` tested this scenario, working as expected.
`Also can we look at refactoring this method a bit.
(Take an overall look at authorizeAndStoreSecret() and storeSecretKey() methods)` addressed with [https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/150/commits/c12cee953b3f619d7d3b96b0b5868d48bed233bc](https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/150/commits/c12cee953b3f619d7d3b96b0b5868d48bed233bc)